libyul/optimiser/RedundantAssignEliminator.cpp

```diff
@@ -239,21 +246,38 @@ void RedundantAssignEliminator::merge(TrackedAssignments& _target, vector<Tracke
 void RedundantAssignEliminator::changeUndecidedTo(YulString _variable, RedundantAssignEliminator::State _newState)
 {
 	for (auto& assignment: m_assignments[_variable])
-		if (assignment.second == State{State::Undecided})
+		if (assignment.second == State::Undecided)
 			assignment.second = _newState;
 }
 
-void RedundantAssignEliminator::finalize(YulString _variable)
+void RedundantAssignEliminator::finalize(YulString _variable, RedundantAssignEliminator::State _finalState)
 {
-	for (auto& assignment: m_assignments[_variable])
+	finalize(m_assignments, _variable, _finalState);
+	for (auto& assignments: m_forLoopInfo.pendingBreakStmts)
+		finalize(assignments, _variable, _finalState);
+	for (auto& assignments: m_forLoopInfo.pendingContinueStmts)
+		finalize(assignments, _variable, _finalState);
+}
+
+void RedundantAssignEliminator::finalize(
+	TrackedAssignments& _assignments,
+	YulString _variable,
+	RedundantAssignEliminator::State _finalState
+)
+{
+	for (auto const& assignment: _assignments[_variable])
 	{
-		assertThrow(assignment.second != State::Undecided, OptimizerException, "");
-		if (assignment.second == State{State::Unused} && MovableChecker{*m_dialect, *assignment.first->value}.movable())
+		State state = assignment.second;
+		if (state == State::Undecided)
+			state = _finalState;
```

Comment:
@chriseth: Actually, just a small thing, as I prefer const/immutable named variables to make it clear at the very beginning that a value of a variable won't change. So here I'd have probably written:
```
State const state = assignment.second == State::Undecided ? _finalState : assignment.second;
```
I believe this would also make the next line below `assertThrow()`) redundant.

What do you think?